components/org.wso2.carbon.identity.application.authenticator.adapter/src/main/java/org/wso2/carbon/identity/application/authenticator/adapter/AbstractAuthenticatorAdapter.java

```diff
@@ -178,4 +201,18 @@ public AuthenticatorPropertyConstants.DefinedByType getDefinedByType() {
 
         return AuthenticatorPropertyConstants.DefinedByType.USER;
     }
+
+    private void handleActionExecutionException(AuthenticationContext context) {
+
+        /* Set a generic error code and message to be sent to the client. An ErrorStatus is generated as the result of
+         the action execution, which will be evaluated and trigger an AuthenticationFailedException in the
+         processAuthenticationResponse method. */
+        String errorCode = "auth_extension_error";
+        String errorMessage = "An error occurred while authenticating user with the external authentication extension.";
```

Comment:
I think we don't need to denote that this failed because of authentication extension which reveals internal detail. May be we can add the display name of the authenticator, else just keep it generic.

"An error occurred while authenticating user with authenticator.getDisplayName."